sources/packages/backend/workflow/test/2022-2023/parttime-assessment/costs/parttime-assessment-2022-2023-costs-child-care-cost.e2e-spec.ts

```diff
@@ -0,0 +1,211 @@
+      "child care costs entered is beyond maximum allowable limit for the " +
+      "given number of dependents, course load and offering weeks.",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder = 1000;
+      assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver = 1000;
+      assessmentConsolidatedData.offeringWeeks = 17;
+      assessmentConsolidatedData.offeringCourseLoad = 20;
+      // Creates 2 eligible dependents.
+      assessmentConsolidatedData.studentDataDependants = [
+        createFakeStudentDependentEligibleForChildcareCost(
+          DependentChildCareEligibility.Eligible0To11YearsOld,
+          assessmentConsolidatedData.offeringStudyStartDate,
+        ),
+        createFakeStudentDependentEligibleForChildcareCost(
+          DependentChildCareEligibility.Eligible12YearsAndOver,
+          assessmentConsolidatedData.offeringStudyStartDate,
+        ),
+      ];
+      const totalChildcareCostEnteredInApplication =
+        assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder +
+        assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver;
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      // Childcare cost for the category 11 years and under must be equal to the value entered in application
+      // for this category as student has one or more dependent in this category.
+      expect(
+        calculatedAssessment.variables.calculatedDataDaycareCosts11YearsOrUnder,
+      ).toBe(assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder);
+      // Childcare cost for the category 12 years and over under must be equal to the value entered in application
+      // for this category as student has one or more dependent in this category.
+      expect(
+        calculatedAssessment.variables.calculatedDataDaycareCosts12YearsOrOver,
+      ).toBe(assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver);
+      // Child care cost expected to be the total sum of values entered in student application.
+      expect(calculatedAssessment.variables.calculatedDataChildCareCost).toBe(
+        totalChildcareCostEnteredInApplication,
+      );
+      // As the total sum of child care costs are beyond the maximum allowable limit
+      // the maximum limit must be considered towards assessed need.
+      expect(
+        calculatedAssessment.variables.calculatedDataTotalChildCareCost,
+      ).toBe(1822.4);
+      expect(
+        calculatedAssessment.variables.calculatedDataTotalChildCareCost,
+      ).toBeLessThan(
+        calculatedAssessment.variables.calculatedDataChildCareCost,
+      );
+    },
+  );
+
```

Comment:
Just a suggestion: Since there are 3 branches below,

<img width="681" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/5f6e34ad-d609-4b7e-ada8-58b47dbcf354">

Maybe we can add another quick test to check offering weeks greater than 34 too!

Reply:
I am not trying to cover every scenario in E2E test.